x-pack/plugins/observability_solution/inventory/public/components/entities_grid/index.tsx

```diff
@@ -136,6 +154,33 @@ export function EntitiesGrid({
     [onChangeSort]
   );
 
+  const getEntityRedirectUrl = useCallback(
+    (entity: LatestEntity) => {
+      const type = entity[ENTITY_TYPE] as EntityType;
+
+      // Any unrecognised types will always return undefined
+      switch (type) {
+        case 'host':
+        case 'container':
+          return assetDetailsLocator?.getRedirectUrl({
+            assetId: entity[ENTITY_DISPLAY_NAME],
+            assetType: type,
+          });
+
+        case 'service': {
+          // For services, the format of the display name is `service.name:service.environment`.
+          // We just want the first part of the name for the locator.
+          const serviceName = entity[ENTITY_DISPLAY_NAME].split(':')[0];
+
+          return serviceOverviewLocator?.getRedirectUrl({
+            serviceName,
```

Comment:
@Bluefinger as you are splitting the string, can you also pass in the service environment, if available?

Reply:
The service overview locator doesn't have the query params for the environment. I could extend that if necessary though

Reply:
Another thing, the service display name string has multiple `:`, so could be `synth-node-1:Synthtrace: infra_hosts_etc`. Is the `environment` part the `Synthtrace` or the whole `Synthtrace: infra_hosts_etc`?

Reply:
The first `:` is what divides the service name and service environment. We append `Synthtrace: ` to the environment when we run synthrace. https://github.com/cauemarcondes/kibana/blob/64f4e74e8634cb8ef06d02a375008690aa33d27e/packages/kbn-apm-synthtrace/src/lib/utils/get_synthtrace_environment.ts#L13-L14

Reply:
Ah I see. Well, that complicates matters in pulling the environment out from the name then, since that adds another `:` to deal with. This might get ugly... 😅

Reply:
as you'll ignore the environement for now, just split the name and use the first element.